xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-preferences/xwiki-platform-notifications-preferences-api/src/main/java/org/xwiki/notifications/preferences/email/NotificationEmailUserPreferenceManager.java

```diff
@@ -36,8 +38,31 @@
     NotificationEmailDiffType getDiffType();
 
     /**
-     * @param userId id of a user
+     * @param userReference reference of a user
      * @return the diff type configured for the given user
+     * @since 14.10
      */
-    NotificationEmailDiffType getDiffType(String userId);
+    default NotificationEmailDiffType getDiffType(DocumentReference userReference)
```

Comment:
You're still not using the `UserReference` type here... See: https://extensions.xwiki.org/xwiki/bin/view/Extension/User%20Module/API/#HBridge